hack/generator/pkg/astmodel/armconversion/arm_conversion_function.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package armconversion
+
+import (
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"go/ast"
+)
+
+// ArmConversionFunction represents an ARM conversion function for converting between an Kubernetes resource
+// and an ARM resource.
+type ArmConversionFunction struct {
```

Comment:
It's not a function though, right? Perhaps this is idiomatic for how you're using Go but it initially seems a bit odd to me...

Reply:
This type in our astmodel represents a code-generated function. This is following a similar naming pattern to the other code-generated function we can attach to a type (`OneOfJSONMarshalFunction`)